e2e/start/basic/app/routes/-server-fns/submit-formdata-fn.tsx

```diff
@@ -0,0 +1,55 @@
+import { createServerFn } from '@tanstack/start'
+
+const testValues = {
+  name: 'Sean',
+}
+
+export const greetUser = createServerFn()
+  .validator((data: FormData) => {
+    if (!(data instanceof FormData)) {
+      throw new Error('Invalid! FormData is required')
+    }
+    const name = data.get('name')
+
+    if (!name) {
+      throw new Error('Name is required')
+    }
+
+    return {
+      name: name.toString(),
+    }
+  })
+  .handler(({ data: { name } }) => {
+    return new Response(`Hello, ${name}!`)
```

Comment:
@schiller-manuel @chorobin could use help fixing the type-level error here not allowing `Response` to be returned. It works at runtime.